aiohttp/payload.py

```diff
@@ -448,7 +457,7 @@ def __init__(self, value: StreamReader, *args: Any, **kwargs: Any) -> None:
 PAYLOAD_REGISTRY.register(BytesIOPayload, io.BytesIO)
 PAYLOAD_REGISTRY.register(
     BufferedReaderPayload, (io.BufferedReader, io.BufferedRandom))
-PAYLOAD_REGISTRY.register(IOBasePayload, io.IOBase)
+PAYLOAD_REGISTRY.register(IOBasePayload, (io.IOBase, TempFilePayload))
```

Comment:
shouldn't this be `BytesIOPayload` and not `IOBasePayload` as the `size` method is available

Reply:
the other issue is that wouldn't it depend on how the file is opened?  a tempfile can be opened as bytes or string